core/cafe/archive.go

```diff
@@ -44,12 +59,22 @@ func (a *Archive) AddFile(blob []byte, fname string) error {
 	return nil
 }
 
+func (a *Archive) VirtualReader() io.Reader {
```

Comment:
`Archive`s can now be "virtual", which just means we're writing direct to a buffer (not a file)... useful for handling pin requests that are actually a directory structure (DAG). In that case, a virtual archive is created from the DAG links, which dumped to the remote pin API.

Reply:
(this happens only when the pinner runs and reaches an entry that is a DAG)